### PR TITLE
improve the method to automatically create components

### PR DESCRIPTION
**Pull request by lferraz**
#### Changes

Fixes # (issue)

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: limbus/core/factory.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 import yaml
 import typeguard
```
Comment (edgarriba) [inline limbus/core/factory.py:9]: What minimum version do we need here ?

Comment (lferraz) [reply limbus/core/factory.py:10]: no idea XD I've not pinned anything yet.

File: tests/core/test_factory.py [hunk 2]
```diff
@@ -30,11 +33,93 @@ def forward(self, inputs: Params) -> ComponentState:  # noqa: D102
...
+    with open(fn, "w") as f:
+        f.write("""
+            torch.unsqueeze:
+                params: {input: torch.Tensor, dim: int}
+                returns: {my_name: torch.Tensor}
+    """)
+    core.register_components_from_yml(str(fn))
+    comp = limbus.components.torch.unsqueeze("test")
+    inp = Params()
+    inp.declare("input", torch.Tensor, torch.randn(2, 3))
```
Comment (edgarriba) [inline tests/core/test_factory.py:74]: why not something like that
```python
inp = Params()
inp.declare("input", torch.Tensor)
inp.declare("dim", int)
comp(torch.randn(2, 3), 2)
```

**Review by edgarriba — APPROVED** [review]
LGTM. Just a small comment

File: tests/core/test_factory.py (hunk 2 above)
Comment (lferraz) [reply tests/core/test_factory.py:74]: my plan is to change the way the params are passed just after closing this branch. We need to think a bit about it, including the mechanism to connect components.

Comment (edgarriba) [reply tests/core/test_factory.py:74]: definitely

**Pull request merged**